modyn/config/schema/pipeline/trigger/ensemble.py

```diff
@@ -0,0 +1,70 @@
+from __future__ import annotations
+
+from typing import Annotated, Callable, ForwardRef, Literal, Union
+
+from modyn.config.schema.base_model import ModynBaseModel
+from pydantic import Field
+
+
+class BaseEnsembleStrategy(ModynBaseModel):
+
+    @property
+    def aggregate_decision_func(self) -> Callable[[dict[str, bool]], bool]:
+        """Returns:
+        Function that aggregates the decisions of the individual triggers."""
+        raise NotImplementedError
+
+
+class MajorityVoteEnsembleStrategy(BaseEnsembleStrategy):
+    id: Literal["MajorityVote"] = Field("MajorityVote")
+
+    @property
+    def aggregate_decision_func(self) -> Callable[[dict[str, bool]], bool]:
+        return lambda decisions: sum(decisions.values()) > len(decisions) / 2
```

Comment:
i am somewhat torn here since we now have business logic in the config files, but this is simple logic so it might be ok...

Reply:
Also considered that. I concluded though that this is the most accessible setup.